libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,220 @@
+            _middlewareSet = middlewareSet ?? new List<IMiddleware>();
+            _userAgent = GetUserAgent();
+        }
+
+        /// <summary>
+        /// Connects the handler to a WebSocket server and begins listening for incoming requests.
+        /// </summary>
+        /// <param name="socket">The socket to use when creating the server.</param>
+        /// <returns>A task that runs until the server is disconnected.</returns>
+        public Task StartAsync(WebSocket socket)
+        {
+            _transportServer = new WebSocketServer(socket, this);
+
+            return _transportServer.StartAsync();
+        }
+
+        /// <summary>
+        /// Connects the handler to a Named Pipe server and begins listening for incoming requests.
+        /// </summary>
+        /// <param name="pipeName">The name of the named pipe to use when creating the server.</param>
+        /// <returns>A task that runs until the server is disconnected.</returns>
+        public Task StartAsync(string pipeName)
+        {
+            _transportServer = new NamedPipeServer(pipeName, this);
+
+            return _transportServer.StartAsync();
+        }
+
+        /// <summary>
+        /// Checks the validity of the request and attempts to map it the correct virtual endpoint,
+        /// then generates and returns a response if appropriate.
+        /// </summary>
+        /// <param name="request">A ReceiveRequest from the connected channel.</param>
+        /// <param name="logger">Optional logger used to log request information and error details.</param>
+        /// <param name="context">Optional context to operate within. Unused in bot implementation.</param>
+        /// /// <param name="cancellationToken">Optional cancellation token.</param>
+        /// <returns>A response created by the BotAdapter to be sent to the client that originated the request.</returns>
+        public override async Task<StreamingResponse> ProcessRequestAsync(ReceiveRequest request, ILogger<RequestHandler> logger, object context = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            logger = logger ?? NullLogger<RequestHandler>.Instance;
+            var response = new StreamingResponse();
+
+            if (request == null || string.IsNullOrEmpty(request.Verb) || string.IsNullOrEmpty(request.Path))
+            {
+                response.StatusCode = (int)HttpStatusCode.BadRequest;
+                logger.LogInformation("Request missing verb and/or path.");
+
+                return response;
+            }
+
+            if (string.Equals(request.Verb, StreamingRequest.GET, StringComparison.InvariantCultureIgnoreCase) &&
+                         string.Equals(request.Path, "/api/version", StringComparison.InvariantCultureIgnoreCase))
+            {
+                response.StatusCode = (int)HttpStatusCode.OK;
+                response.SetBody(new VersionInfo() { UserAgent = _userAgent });
```

Comment:
P1 - A Get just returns an OK? 

If the user is request /api/version should we consider sending back more than just the Agent string?

Reply:
At the moment this is only serving the purpose of providing the user agent for the sake of not breaking our telemetry tools. I think it will likely expand in the future as we decide what additional information it should be providing, as part of a bigger discussion about protocol versioning and negotiation.